cylc/flow/async_util.py

```diff
@@ -17,6 +17,7 @@
 
 import asyncio
 from functools import partial, wraps
+from inspect import signature
```

Comment:
Did you look at `functools.wraps`?

Reply:
Yes but couldn't figure out how to get it to play along with this:

https://github.com/MetRonnie/cylc-flow/blob/67e7cf9c07eaa2f59b244fa99f8db338d4040984/cylc/flow/async_util.py#L379-L392